<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>NGSeasy by KHP-Informatics</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">NGSeasy</h1>
      <h2 class="project-tagline">Dockerised Next Generation Sequencing Pipeline (QC, Align, Calling, Annotation)</h2>
      <a href="https://github.com/KHP-Informatics/ngseasy" class="btn">View on GitHub</a>
      <a href="https://github.com/KHP-Informatics/ngseasy/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/KHP-Informatics/ngseasy/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <p><img src="https://github.com/KHP-Informatics/ngseasy/blob/sjn_dev/figs/NGSeasy_logo_0.0.1.png" alt="NGSeasy_logo"></p>

<h1>
<a id="ngseasy-beta-a-dockerized-ngs-pipeline-and-tool-box" class="anchor" href="#ngseasy-beta-a-dockerized-ngs-pipeline-and-tool-box" aria-hidden="true"><span class="octicon octicon-link"></span></a>NGSeasy (beta): A <a href="https://www.docker.com/">Dockerized</a> NGS pipeline and tool-box</h1>

<p><strong>With NGSeasy you can now have full suite of NGS tools up and running on any high end workstation in an afternoon</strong>  </p>

<p><strong>Authors:</strong> Stephen J Newhouse and Amos Folarin<br>
<strong>Release Version:</strong> <strong>1.0-r001</strong><br>
<strong>Release:</strong> dirty_tango<br>
<strong>Publication:</strong> Folarin AA, Dobson RJ and Newhouse SJ. <a href="http://f1000research.com/articles/4-997/v1">NGSeasy</a>: a next generation sequencing pipeline in Docker containers [version 1; referees: awaiting peer review] F1000Research 2015, 4(ISCB Comm J):997 (doi: 10.12688/f1000research.7104.1).   </p>

<ul>
<li>
<strong>NGSeasy-1.0 Full Production release will be available Late 2015</strong><br>
</li>
<li><strong>NGSeasy-1.0-r001 (dirty_tango) contains most of the core functionality to go from raw fastq to raw vcf calls</strong></li>
<li><strong>NGSeasy will update every 12 months</strong></li>
<li><strong>GUI in development</strong></li>
<li><strong>Lets us know if you want other tools added to NGSeasy</strong></li>
</ul>

<p><a href="https://github.com/KHP-Informatics/ngseasy.git"><strong>NGSeasy is completely open source and we encourage interested folks to jump in and get involved in the dev with us.</strong></a></p>

<h2>
<a id="ngseasy-genome-comparison--analytic-testing-gcat-reports" class="anchor" href="#ngseasy-genome-comparison--analytic-testing-gcat-reports" aria-hidden="true"><span class="octicon octicon-link"></span></a>NGSeasy: Genome Comparison &amp; Analytic Testing (<a href="http://www.bioplanet.com/gcat">GCAT</a>) Reports</h2>

<p>Here we provide a quick look at basic NGSeasy performance (more results coming soon).</p>

<table>
<thead>
<tr>
<th>GCAT Report</th>
<th>Test Data</th>
<th>Pipeline</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="http://www.bioplanet.com/gcat/reports/6167-seeirhwtfp/variant-calls/illumina-100bp-pe-exome-150x/ngseasy-ntrim-bwa-freebayes-d/compare-570-270-181/group-read-depth"><strong>NGSEASY-NTRIM-BWA-FREEBAYES-D</strong></a></td>
<td><a href="https://s3-eu-west-1.amazonaws.com/ngseasy.data/fastq_test_data/illumina.100bp.pe.wex.150x_1.fastq.gz">illumina-100bp-pe-exome-150x</a></td>
<td><a href="https://github.com/KHP-Informatics/ngseasy/blob/master/test/ngseasy_test.config.freebayes.tsv">fastq &gt; bwa &gt; freebayes</a></td>
</tr>
<tr>
<td><a href="http://www.bioplanet.com/gcat/reports/6264-uzdswfzmkv/variant-calls/illumina-100bp-pe-exome-150x/ngseasy-ntrim-bwa-platypus-d/compare-571-180/group-read-depth"><strong>NGSEASY-NTRIM-BWA-PLATYPUS-D</strong></a></td>
<td><a href="https://s3-eu-west-1.amazonaws.com/ngseasy.data/fastq_test_data/illumina.100bp.pe.wex.150x_2.fastq.gz">illumina-100bp-pe-exome-150x</a></td>
<td><a href="https://github.com/KHP-Informatics/ngseasy/blob/master/test/ngseasy_test.config.platypus.tsv">fastq &gt; bwa &gt; platypus</a></td>
</tr>
</tbody>
</table>

<p>An example of the run commands:</p>

<div class="highlight highlight-source-shell"><pre>ngseasy -c ngseasy_test.config.freebayes.tsv -d /media/Data/ngs_projects
ngseasy -c ngseasy_test.config.platypus.tsv  -d /media/Data/ngs_projects</pre></div>

<h2>
<a id="author-contact-details" class="anchor" href="#author-contact-details" aria-hidden="true"><span class="octicon octicon-link"></span></a>Author Contact Details</h2>

<p>Please contact us for help/guidance on using the beta release.</p>

<table>
<thead>
<tr>
<th>Author</th>
<th>email</th>
<th>Twitter</th>
<th>LinkedIn</th>
</tr>
</thead>
<tbody>
<tr>
<td>Dr Stephen J Newhouse</td>
<td><a href="mailto:stephen.j.newhouse@gmail.com">stephen.j.newhouse@gmail.com</a></td>
<td><a href="https://twitter.com/s_j_newhouse?lang=en">@s_j_newhouse</a></td>
<td>View Steve's profile on <a href="http://uk.linkedin.com/pub/dr-stephen-newhouse/29/89a/11a">LinkedIn</a>
</td>
</tr>
<tr>
<td>Dr Amos Folarin</td>
<td><a href="mailto:amosfolarin@gmail.com">amosfolarin@gmail.com</a></td>
<td><a href="https://twitter.com/amosfolarin?lang=en">@amosfolarin</a></td>
<td>View Amos's profile on <a href="http://www.linkedin.com/pub/amos-folarin/34/b06/978">LinkedIn</a>
</td>
</tr>
</tbody>
</table>

<h2>
<a id="issues-questions-and-queries" class="anchor" href="#issues-questions-and-queries" aria-hidden="true"><span class="octicon octicon-link"></span></a>Issues, Questions and Queries</h2>

<p><strong>Please Direct all queries to [<a href="https://github.com/KHP-Informatics/ngseasy/issues">https://github.com/KHP-Informatics/ngseasy/issues</a>]</strong></p>

<p>When sending bug reports etc please provide:-</p>

<ul>
<li>Date of Download</li>
<li>OS and version</li>
<li>Basic Machine Specs (CPU, RAM)</li>
<li>Network Speed (<a href="https://github.com/KHP-Informatics/ngseasy/tree/master#recommended-network-speed">Testing Internet Connection Speed</a>)</li>
<li>The Code you ran eg:- <code>ngseasy -c my.config.tsv -d /My/Dir</code>
</li>
<li>your config file<br>
</li>
<li>The exact error as printed to screen</li>
</ul>

<p><strong>WARNING!</strong>
NGSeasy is not <a href="http://www.urbandictionary.com/define.php?term=Numpty">numpty</a> or bad data proof!  </p>

<p><strong>Please read the docs</strong>, stay calm, take your time and think about what you are doing...and if [<a href="http://www.google.com">www.google.com</a>] doesnt help, then please direct all queries to [<a href="https://github.com/KHP-Informatics/ngseasy/issues">https://github.com/KHP-Informatics/ngseasy/issues</a>].</p>

<h2>
<a id="install-docker" class="anchor" href="#install-docker" aria-hidden="true"><span class="octicon octicon-link"></span></a>Install Docker</h2>

<p>Full instructions at <a href="https://docs.docker.com/">https://docs.docker.com/</a>.</p>

<p>Some fixes to make life easy...allows you to run <code>docker</code> without <code>sudo</code>.</p>

<p>This may differ for your OS, and mostly applies to flavours of <code>Linux</code>. Check with your sys admin or just Google <a href="https://www.google.com">https://www.google.com</a>.</p>

<p>MAC/Windows users using <a href="http://boot2docker.io/">http://boot2docker.io/</a> should be fine. Read the docs or just Google <a href="https://www.google.com">https://www.google.com</a>.</p>

<p><strong>Create a docker group</strong></p>

<div class="highlight highlight-source-shell"><pre>sudo addgroup docker</pre></div>

<p><strong>Add user to docker group</strong>  </p>

<p>Here user is <code>ec2-user</code></p>

<div class="highlight highlight-source-shell"><pre>sudo usermod -aG docker ec2-user</pre></div>

<p>Log out and log back in.</p>

<p>This ensures your user is running with the correct permissions.</p>

<p>Verify your work by running <code>docker</code> without <code>sudo</code>.</p>

<div class="highlight highlight-source-shell"><pre>docker run hello-world</pre></div>

<p>..this is what you should get...</p>

<pre><code>Unable to find image 'hello-world:latest' locally
Pulling repository hello-world
91c95931e552: Download complete
a8219747be10: Download complete
Status: Downloaded newer image for hello-world:latest
Hello from Docker.
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (Assuming it was not already locally available.)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

For more examples and ideas, visit:
 http://docs.docker.com/userguide/
</code></pre>

<h2>
<a id="docker-security" class="anchor" href="#docker-security" aria-hidden="true"><span class="octicon octicon-link"></span></a>Docker Security...</h2>

<p>This post reviews the various security implications of using Docker to run applications within containers, and how to address them:
<a href="https://blog.docker.com/2013/08/containers-docker-how-secure-are-they/#more-697">How Secure are Containers?</a></p>

<blockquote>
<p>Docker containers are, by default, quite secure; especially if you take care of running your processes inside the containers as non-privileged users (i.e. non root).</p>
</blockquote>

<h2>
<a id="get-ngseasy" class="anchor" href="#get-ngseasy" aria-hidden="true"><span class="octicon octicon-link"></span></a>Get NGSeasy</h2>

<div class="highlight highlight-source-shell"><pre>
<span class="pl-c">#############################################</span>
<span class="pl-c">## Get NGSeasy                             ##</span>
<span class="pl-c">#############################################</span>

<span class="pl-c1">cd</span> /home/<span class="pl-smi">${USER}</span>

git clone https://github.com/KHP-Informatics/ngseasy.git</pre></div>

<h2>
<a id="install-ngseasy" class="anchor" href="#install-ngseasy" aria-hidden="true"><span class="octicon octicon-link"></span></a>Install NGSeasy</h2>

<ul>
<li>Default install directory is <code>/home/${USER}</code>
</li>
<li>in this example user home is <code>/home/ec2-user</code>
</li>
<li>
<code>make INTSALLDIR="/home/ec2-user" all</code>

<ul>
<li>sets up top level directory structure</li>
<li>gets all docker images</li>
<li>gets indexed hg19 and b37 genomes</li>
<li>gets GATK recources for hg19 and b37 genomes</li>
<li>gets whole genome and exome test data</li>
</ul>
</li>
<li>Always set your <strong><code>INTSALLDIR</code></strong> : If you run <code>sudo make all</code> the install path will be <code>/home/root</code>. Please dont do this!</li>
<li>
<code>sudo make install</code> installs scripts to <code>/usr/local/bin/</code><br>
</li>
</ul>

<div class="highlight highlight-source-shell"><pre><span class="pl-c">#############################################</span>
<span class="pl-c">## install NGSeasy                         ##</span>
<span class="pl-c">#############################################</span>

<span class="pl-c1">cd</span> ngseasy

<span class="pl-c">## 1.</span>
make INTSALLDIR=<span class="pl-s"><span class="pl-pds">"</span>/home/ec2-user<span class="pl-pds">"</span></span> all

<span class="pl-c">## 2.</span>
sudo make install</pre></div>

<p>Installation can take a while, 1-2 hours, so go get a coffee../just chill...if your network is bad...then who knows how long...still..just chill...or go get fast internet!</p>

<h2>
<a id="ngseasy-security" class="anchor" href="#ngseasy-security" aria-hidden="true"><span class="octicon octicon-link"></span></a>NGSeasy Security</h2>

<p><strong>All NGSeasy applications are run as the non-root user <code>pipeman</code> within each container</strong></p>

<h2>
<a id="recommended-network-speed" class="anchor" href="#recommended-network-speed" aria-hidden="true"><span class="octicon octicon-link"></span></a>Recommended Network Speed</h2>

<p><strong>&gt; 500 Mbit/s</strong> : anything less will add a lot of time to set up (days - weeks).</p>

<p><strong>Testing Internet Connection Speed</strong></p>

<p>source : <a href="http://askubuntu.com/questions/104755/how-to-check-internet-speed-via-terminal">http://askubuntu.com/questions/104755/how-to-check-internet-speed-via-terminal</a></p>

<div class="highlight highlight-source-shell"><pre>wget -O speedtest-cli https://raw.github.com/sivel/speedtest-cli/master/speedtest_cli.py
chmod +x speedtest-cli
./speedtest-cli</pre></div>

<pre><code>Retrieving speedtest.net configuration...
Retrieving speedtest.net server list...
Testing from Comcast Cable (x.x.x.x)...
Selecting best server based on ping...
Hosted by FiberCloud, Inc (Seattle, WA) [12.03 km]: 44.028 ms
Testing download speed........................................
Download: 32.29 Mbit/s
Testing upload speed..................................................
Upload: 5.18 Mbit/s
</code></pre>

<h2>
<a id="install-time-on-amazon-ec2" class="anchor" href="#install-time-on-amazon-ec2" aria-hidden="true"><span class="octicon octicon-link"></span></a>Install time on Amazon EC2</h2>

<p>Connection Speed: ~ 800 Mbit/s</p>

<pre><code>real    94m54.237s
user    12m26.960s
sys     28m46.648s
</code></pre>

<p>Note: We have only tested NGSeasy installation on Amazon EC2, Openstack and UK University Networks. These are all fairly fast networks with speeds exceeding 800 Mbit/s on average.</p>

<h2>
<a id="running-ngseasy-for-the-first-time-on-the-test-data" class="anchor" href="#running-ngseasy-for-the-first-time-on-the-test-data" aria-hidden="true"><span class="octicon octicon-link"></span></a>Running NGSeasy for the first time on the test data</h2>

<p><strong>Important!</strong> NGSeasy is controlled from a single <code>config</code> file. See <a href="https://github.com/KHP-Informatics/ngseasy/blob/master/test/ngseasy_test.config.tsv">ngseasy_test.config.tsv</a> for a basic template. It is important that the user sets this up properly before running NGSeasy.</p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c">#############################################</span>
<span class="pl-c">## 0. Move to config file dir</span>

<span class="pl-c1">cd</span> /home/ec2-user/ngs_projects/config_files/

<span class="pl-c">#############################################</span>
<span class="pl-c">## 1. Run basic test</span>

ngseasy -c ngseasy_test.config.tsv -d /home/ec2-user/ngs_projects</pre></div>

<h3>
<a id="what-should-happen" class="anchor" href="#what-should-happen" aria-hidden="true"><span class="octicon octicon-link"></span></a>What should happen...</h3>

<p>This runs the following basic pipeline on Whole Exome PE 30x Illumina data, aligning to b37 (in theory...give it a try).  </p>

<ul>
<li>FastQC &gt; Trimmomatic &gt; BWA &gt; Platypus</li>
</ul>

<h3>
<a id="some-notes-and-pointers" class="anchor" href="#some-notes-and-pointers" aria-hidden="true"><span class="octicon octicon-link"></span></a>Some notes and pointers</h3>

<ul>
<li>Edit <strong>NCPU</strong> in  <strong>[ngseasy_test.config.tsv]</strong> to suit your system<br>
</li>
<li>Edit <strong>PROJECT_DIR</strong> in  <strong>[ngseasy_test.config.tsv]</strong> to suit your install path</li>
<li>We expect the user to palce all raw fastq files in <code>raw_fastq</code>. NGSeasy uses this
as a stagging area for new project and sample data.<br>
</li>
<li>right now, always run <code>ngseasy</code> from the location/directory that contains the config.file<br>
</li>
<li>each component of ngseasy can be run as a standalone script<br>
</li>
</ul>

<hr>

<h1>
<a id="ngseasy-easy-analysis-of-next-generation-sequencing" class="anchor" href="#ngseasy-easy-analysis-of-next-generation-sequencing" aria-hidden="true"><span class="octicon octicon-link"></span></a>NGSeasy (Easy Analysis of Next Generation Sequencing)</h1>

<p>We present <strong>NGSeasy (Easy Analysis of Next Generation Sequencing)</strong>, a flexible and easy-to-use NGS pipeline for automated alignment, quality control, variant calling and annotation. The pipeline allows users with minimal computational/bioinformatic skills to set up and run an NGS analysis on their own samples, in less than an afternoon, on any operating system (Windows, iOS or Linux) or infrastructure (workstation, cluster or cloud).</p>

<p>NGS pipelines typically utilize a large and varied range of software components and incur a substantial configuration burden during deployment which limits their portability to different computational environments. NGSeasy simplifies this by providing the pipeline components encapsulated in Docker™ containers and bundles in a wide choice of tools for each module. Each module of the pipeline represents one functional grouping of tools (e.g. sequence alignment, variant calling etc.).</p>

<p>Deploying the pipeline is as simple as pulling the container images from the public repository into any host running Docker. NGSeasy can be deployed on any medium to high-end workstation, high performance computer cluster and compute clouds (public/private cloud computing) - enabling instant access to elastic scalability without investment overheads for additional compute hardware and makes open and reproducible research straight forward for the greater scientific community.</p>

<h3>
<a id="advantages" class="anchor" href="#advantages" aria-hidden="true"><span class="octicon octicon-link"></span></a>Advantages</h3>

<ul>
<li>Easy to use for non-informaticians.<br>
</li>
<li>All run from a single config file that can be made in Excel.<br>
</li>
<li>User can select from mutiple aligners, variant callers and variant annotators</li>
<li>No scary python, .yaml or .json files...just one simple Excel workbook saved as a textfile.<br>
</li>
<li>Just follow our simple set of instructions and NGS away!<br>
</li>
<li>Choice of aligners and variant callers and anntators<br>
</li>
<li>Allows reproducible research<br>
</li>
<li>Version controlled for auditing<br>
</li>
<li>Customisable<br>
</li>
<li>Easy to add new tools<br>
</li>
<li>If it's broke...we will fix it..</li>
<li>Enforced naming convention and directory structures<br>
</li>
<li>Allows users to run "Bake Offs" between tools with ease<br>
</li>
</ul>

<p>We have adapted the current best practices from the Genome Analysis Toolkit (GATK, <a href="http://www.broadinstitute.org/gatk/guide/best-practices">http://www.broadinstitute.org/gatk/guide/best-practices</a>)  for processing raw alignments in SAM/BAM format and variant calling. The current workflow, has been optimised for Illumina platforms, but can easily be adapted for other sequencing platforms, with minimal effort.  </p>

<p>As the containers themselves can be run as executables with pre-specified cpu and RAM resources, the orchestration of the pipeline can be placed under the control of conventional load balancers if this mode is required.  </p>

<h2>
<a id="genomes" class="anchor" href="#genomes" aria-hidden="true"><span class="octicon octicon-link"></span></a>Genomes</h2>

<table>
<thead>
<tr>
<th>Genome Build</th>
</tr>
</thead>
<tbody>
<tr>
<td>hs37d5</td>
</tr>
<tr>
<td>b37</td>
</tr>
<tr>
<td>hg19</td>
</tr>
<tr>
<td>hs38DH</td>
</tr>
</tbody>
</table>

<hr>

<h1>
<a id="overview-of-the-ngseasy-pipeline-components" class="anchor" href="#overview-of-the-ngseasy-pipeline-components" aria-hidden="true"><span class="octicon octicon-link"></span></a>Overview of the NGSeasy Pipeline Components</h1>

<p>The basic pipeline contains all the basic tools needed for manipulation and
quality control of raw fastq files (ILLUMINA focused), SAM/BAM manipulation,
alignment, cleaning (based on GATK best practises [<a href="http://www.broadinstitute.org/gatk/guide/best-practices">http://www.broadinstitute.org/gatk/guide/best-practices</a>]) and first pass
variant discovery. Separate containers are provided for indepth variant annotation,
structural variant calling, basic reporting and visualisations.  </p>

<p><img src="https://github.com/KHP-Informatics/ngs/blob/dev2/figs/ngsEASY_component_visualisation.png" alt="ngsEASY"></p>

<h2>
<a id="a-special-note-on-the-ngseasy-base-image" class="anchor" href="#a-special-note-on-the-ngseasy-base-image" aria-hidden="true"><span class="octicon octicon-link"></span></a>A Special note on the NGSeasy base image.</h2>

<p>We include the following - what we think of as - <strong><em>NGS Powertools</em></strong> in the <strong><a href="">compbio/ngseasy-base</a></strong> image.
These are all tools that allow the user to slice and dice BED/SAM/BAM/VCF files in multiple ways.</p>

<ol>
<li> samtools<br>
</li>
<li> bcftools<br>
</li>
<li> vcftools<br>
</li>
<li> vcflib<br>
</li>
<li> bamUtil<br>
</li>
<li> bedtools2<br>
</li>
<li> ogap<br>
</li>
<li> samblaster<br>
</li>
<li> sambamba<br>
</li>
<li>bamleftalign<br>
</li>
<li>seqtk<br>
</li>
<li>parallel<br>
</li>
</ol>

<p>This image is used as the base of all our compbio/ngseasy-* tools.</p>

<p><strong>Why not a separate containers per application?</strong> The more docker-esque approach, would be to have separate containers for each NGS tool. However, this belies the fact that many of these tools interact in a deep way. Therefore, we built  these into a single development environment for ngseasy, to allow pipes and streamlined system calls for manipulating the output of NGS pipelines (BED/SAM/BAM/VCF files).</p>

<hr>

<h1>
<a id="the-full-ngseasy-pipeline" class="anchor" href="#the-full-ngseasy-pipeline" aria-hidden="true"><span class="octicon octicon-link"></span></a>The Full NGSeasy pipeline</h1>

<p>The NGSeasy pipelines implement the following :-</p>

<ul>
<li><p><strong>Quality control of raw fastq</strong> files using <strong><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FASTQC</a></strong>  </p></li>
<li><p><strong>Read trimming</strong> using <strong><a href="http://www.usadellab.org/cms/?page=trimmomatic">TRIMMOMATIC</a></strong>.</p></li>
<li>
<p><strong>Alignment</strong> using one of</p>

<ul>
<li>
<strong><a href="http://bio-bwa.sourceforge.net/">BWA</a></strong><br>
</li>
<li><strong><a href="http://www.well.ox.ac.uk/project-stampy">STAMPY</a></strong></li>
<li>
<strong><a href="http://www.novocraft.com">NOVOALIGN</a></strong><br>
</li>
<li>
<strong><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">BOWTIE2</a></strong><br>
</li>
<li>
<strong><a href="http://snap.cs.berkeley.edu/">SNAP</a></strong><br>
</li>
</ul>
</li>
<li><p><strong>SAM/BAM sorting and indexing</strong> with <strong><a href="https://github.com/lomereiter/sambamba">SAMBAMBA</a></strong>.  </p></li>
<li><p><strong>Read Group information added</strong> using <strong><a href="http://broadinstitute.github.io/picard/">PICARDTOOLS</a>:<a href="http://broadinstitute.github.io/picard/command-line-overview.html#AddOrReplaceReadGroups">AddOrReplaceReadGroups</a></strong></p></li>
<li><p><strong>Duplicate marking</strong> with <strong><a href="https://github.com/GregoryFaust/samblaster">SAMBLASTER</a></strong>.  </p></li>
</ul>

<blockquote>
<p>For academic users and/or commercial/clinical groups whom have paid for GATK licensing, the next steps are to perform</p>
</blockquote>

<ul>
<li>
<strong>Indel indel realignment and base quality score recalibration</strong> using <strong><a href="https://www.broadinstitute.org/gatk/">GATK</a></strong> built in tools :

<ul>
<li><strong><a href="https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.php">GATK:RealignerTargetCreator</a></strong></li>
<li><strong><a href="https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.php">GATK:IndelRealigner</a></strong></li>
<li><strong><a href="https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.php">GATK:BaseRecalibrator</a></strong></li>
</ul>
</li>
</ul>

<blockquote>
<p>For the non-GATK version</p>
</blockquote>

<ul>
<li>
<p><strong>Base quality score recalibration</strong> using <a href="http://genome.sph.umich.edu/wiki/BamUtil">BamUtil</a>  </p>

<ul>
<li><strong><a href="http://genome.sph.umich.edu/wiki/BamUtil:_recab">BamUtil:recab</a></strong></li>
</ul>
</li>
<li>
<p><strong>Post alignment quality control and reporting</strong> is performed usng a number of tools and custom scripts:</p>

<ul>
<li><strong><a href="https://github.com/samtools/samtools">SAMTOOLS:flagstats</a></strong></li>
<li><strong><a href="https://github.com/arq5x/bedtools2">BEDTOOLS:genomecov</a></strong></li>
<li><strong><a href="https://github.com/arq5x/bedtools2">BEDTOOLS:bamtobed</a></strong></li>
<li><strong><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics">PICARDTOOLS:CollectMultipleMetrics</a></strong></li>
<li><strong><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectAlignmentSummaryMetrics">PICARDTOOLS:CollectAlignmentSummaryMetrics</a></strong></li>
<li><strong><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectWgsMetrics">PICARDTOOLS:CollectWgsMetrics</a></strong></li>
<li>
<strong><a href="http://broadinstitute.github.io/picard/command-line-overview.html#CollectTargetedPcrMetrics">PICARDTOOLS:CollectTargetedPcrMetrics</a></strong> (coming soon)</li>
</ul>
</li>
<li>
<p><strong>SNP and small INDEL</strong> calling using one of the following or a combibation of these tools, if the <code>ensemble</code> method is called using <strong><a href="https://github.com/chapmanb/bcbio.variation">bcbio.variation variant-ensemble</a></strong></p>

<ul>
<li><strong><a href="https://github.com/ekg/freebayes">FREEBAYES</a></strong></li>
<li><strong><a href="http://www.well.ox.ac.uk/platypus">PLATYPUS</a></strong></li>
<li><strong><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.php">GATK:UnifiedGenotyper</a></strong></li>
<li><strong><a href="https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php">GATK:HaplotypeCaller</a></strong></li>
</ul>
</li>
<li>
<p><strong>Structural Variant (CNV)</strong> calling using one of the following or or a combibation of if the <code>ensemble</code> methods are called:-</p>

<ul>
<li><strong><a href="https://github.com/tobiasrausch/delly">DELLY</a> : in Dev</strong></li>
<li><strong><a href="https://github.com/arq5x/lumpy-sv/">LUMPY</a>: in Dev</strong></li>
<li><strong><a href="http://www.bioinf.jku.at/software/cnmops/">cn.MOPS</a>: in Dev</strong></li>
<li><strong><a href="https://www.stt.msu.edu/users/hengwang/mHMM.html">m-HMM</a>: in Dev</strong></li>
<li><strong><a href="http://cran.r-project.org/web/packages/ExomeDepth/index.html">ExomeDepth</a>: in Dev</strong></li>
<li>
<strong><a href="http://www-genepi.med.utah.edu/suppl/SLOPE/index.html">SLOPE</a>: in Dev and not tested</strong><br>
</li>
<li>
<strong><a href="http://sv.gersteinlab.org/">cnvnator</a>: in Dev</strong><br>
</li>
</ul>
</li>
<li>
<p><strong>Variant annotation</strong> using using one of the following or a combibation of if the <code>ensemble</code> methods are called.</p>

<ul>
<li><strong><a href="http://snpeff.sourceforge.net/">SnpEff</a>: in Dev</strong></li>
<li><strong><a href="http://www.openbioinformatics.org/annovar/">ANNOVAR</a>: in Dev</strong></li>
<li><strong><a href="http://www.ensembl.org/info/docs/tools/vep/index.html">VEP</a>: in Dev</strong></li>
</ul>
</li>
<li><p><strong>Variant reporting</strong> using custom scripts</p></li>
</ul>

<p><strong>Note</strong> Some of the later functions i.e. variant annotation and qc reporting are still in dev.  </p>

<p><strong>We highly recommed read trimming prior to alignment.</strong>
We have noticed considerable speed-ups in alignmnet time and increased quality of SNP/INDEL calls using trimmed vs raw fastq.  </p>

<p><strong>Base quality score recalibration is also recommended.</strong><br>
As an alternative to GATK, we have added fucntionality for use of
<strong><a href="https://github.com/statgen/bamUtil">BamUtil</a>:<a href="http://genome.sph.umich.edu/wiki/BamUtil:_recab">recab</a></strong>
for base quality score recalibration.</p>

<p><strong>Non-GATK users</strong></p>

<ul>
<li>are encouraged to use aligners such as <strong><a href="http://www.well.ox.ac.uk/project-stampy">stampy</a></strong> and <strong><a href="http://www.novocraft.com">novoalign</a></strong> that perform base quality score recal on the fly.<br>
</li>
<li>are encouraged to use variant callers that perform local re-aligmnet around candidate sites to mitigate the need for the indel realignment stages.<br>

<ul>
<li><strong><a href="https://github.com/ekg/freebayes">freebayes</a></strong></li>
<li><strong><a href="http://www.well.ox.ac.uk/platypus">platypus</a></strong></li>
</ul>
</li>
</ul>

<hr>

<h2>
<a id="dockerised-ngs-tools" class="anchor" href="#dockerised-ngs-tools" aria-hidden="true"><span class="octicon octicon-link"></span></a>Dockerised NGS Tools</h2>

<p>All NGSeasy Docker images can be pulled down from <strong><a href="https://hub.docker.com/u/compbio/">compbio Docker Hub</a></strong> or using the Makefile.<br>
We provide an Amazon EBS data volume with indexed genomes: XXXXXX  </p>

<h2>
<a id="table-1-ngseasy-tools" class="anchor" href="#table-1-ngseasy-tools" aria-hidden="true"><span class="octicon octicon-link"></span></a>Table 1. NGSeasy Tools</h2>

<table>
<thead>
<tr>
<th>Docker Image</th>
<th>Version</th>
<th>NGS Tool (version)</th>
<th>Short Description</th>
<th>URL</th>
</tr>
</thead>
<tbody>
<tr>
<td>compbio/ngseasy-base</td>
<td>1.0-r001</td>
<td>VCFtools (v0.1.12b)</td>
<td>manipulate vcf</td>
<td><a href="http://vcftools.sourceforge.net/">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>vt (latest)</td>
<td>manipulate vcf</td>
<td><a href="http://genome.sph.umich.edu/wiki/Vt">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>bcftools (1.2-5-g7fa0d25)</td>
<td>manipulate vcf</td>
<td><a href="https://github.com/samtools/bcftools">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>vcflib (v1.0.0)</td>
<td>manipulate vcf</td>
<td><a href="https://github.com/ekg/vcflib">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>samtools (1.2-17-ge91985a)</td>
<td>manipulate sam/bam</td>
<td><a href="https://github.com/samtools/">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>samblaster (0.1.21)</td>
<td>manipulate sam/bam</td>
<td><a href="https://github.com/GregoryFaust/samblaster">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>sambamba (v0.5.1)</td>
<td>manipulate sam/bam</td>
<td><a href="https://github.com/lomereiter/sambamba">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>bamUtil (1.0.13)</td>
<td>manipulate sam/bam</td>
<td><a href="https://github.com/statgen/bamUtil">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>bedtools (v2.23.0-10-g447cb97)</td>
<td>manipulate bed files</td>
<td><a href="https://github.com/arq5x/bedtools2">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>seqtk (1.0-r77-dirty)</td>
<td>manipulate fastq</td>
<td><a href="https://github.com/lh3/seqtk">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>vawk (0.0.2)</td>
<td>manipulate vcf</td>
<td><a href="https://github.com/cc2qe/vawk">link</a></td>
</tr>
<tr>
<td>-</td>
<td>-</td>
<td>bioawk (latest)</td>
<td>manipulate sam/bam/vcf</td>
<td><a href="https://github.com/lh3/bioawk">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-fastqc</td>
<td>1.0-r001</td>
<td>fastqc (v0.11.2)</td>
<td>FASTQ Quality Control Plots</td>
<td><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-trimmomatic</td>
<td>1.0-r001</td>
<td>trimmomatic (0.32)</td>
<td>FASTQ Quality Trimming</td>
<td><a href="http://www.usadellab.org/cms/index.php?page=trimmomatic">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-bwa</td>
<td>1.0-r001</td>
<td>bwa ( 0.7.12-r1039)</td>
<td>Aligner</td>
<td><a href="http://bio-bwa.sourceforge.net/">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-stampy</td>
<td>1.0-r001</td>
<td>stampy (stampy-1.0.27)</td>
<td>Aligner</td>
<td><a href="http://www.well.ox.ac.uk/project-stampy">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-snap</td>
<td>1.0-r001</td>
<td>snap-aligner (1.0beta.18)</td>
<td>Aligner</td>
<td><a href="http://snap.cs.berkeley.edu/">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-bowtie2</td>
<td>1.0-r001</td>
<td>bowtie2 (2.2.4)</td>
<td>Aligner</td>
<td><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-novoalign</td>
<td>1.0-r001</td>
<td>novoalign (3.02.13)</td>
<td>Aligner</td>
<td><a href="http://www.novocraft.com/products/novoalign/">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-gatk</td>
<td>1.0-r001</td>
<td>gatk (3.4-0)</td>
<td>NGS PowerTools</td>
<td><a href="https://www.broadinstitute.org/gatk/">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-picardtools</td>
<td>1.0-r001</td>
<td>picardtools (1.128)</td>
<td>NGS PowerTools</td>
<td><a href="http://broadinstitute.github.io/picard/">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-glia</td>
<td>1.0-r001</td>
<td>glia (latest)</td>
<td>NGS local realignment</td>
<td><a href="https://github.com/ekg/glia">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-platypus</td>
<td>1.0-r001</td>
<td>platypus (0.8.1)</td>
<td>Variant Caller</td>
<td><a href="https://github.com/andyrimmer/Platypus">link</a></td>
</tr>
<tr>
<td>compbio/ngseasy-freebayes</td>
<td>1.0-r001</td>
<td>freebayes (v0.9.21-19-gc003c1e)</td>
<td>Variant Caller</td>
<td><a href="https://github.com/ekg/freebayes">link</a></td>
</tr>
</tbody>
</table>

<h2>
<a id="running-an-ngseasy-tool-interactively" class="anchor" href="#running-an-ngseasy-tool-interactively" aria-hidden="true"><span class="octicon octicon-link"></span></a>Running an NGSeasy Tool Interactively</h2>

<p>Run as non-root user <code>pipeman</code>.</p>

<p><code>-v /media/Data:/home/pipeman</code> : Mounts local directory <code>/media/Data</code> to container directory <code>/home/pipeman</code></p>

<div class="highlight highlight-source-shell"><pre>TOOL=<span class="pl-s"><span class="pl-pds">"</span>bwa<span class="pl-pds">"</span></span>

docker run \
-P \
-w /home/pipeman \
-e HOME=/home/pipeman \
-e USER=pipeman \
--user pipeman \
-v /media/Data:/home/pipeman \
-it compbio/ngseasy-<span class="pl-smi">${TOOL}</span>:1.0 /bin/bash</pre></div>

<hr>

<h1>
<a id="dockerised-ngseasy" class="anchor" href="#dockerised-ngseasy" aria-hidden="true"><span class="octicon octicon-link"></span></a>Dockerised NGSeasy</h1>

<p><img src="https://github.com/KHP-Informatics/ngs/blob/master/figs/Docker_container_engine_logo.png" alt="docker">  </p>

<p>The following section describes getting the Dockerised NGSeasy Pipeline(s) and Resources, project set up and running NGSeasy.</p>

<p>Getting all resources and building required tools will take a few hours depending on network connections and any random "ghosts in the machine" - half a day in reality.
But once you're set up, thats it - you are good to go.</p>

<h2>
<a id="system-requirements" class="anchor" href="#system-requirements" aria-hidden="true"><span class="octicon octicon-link"></span></a>System Requirements</h2>

<p>See Table <a href=""><strong>System Requirements</strong></a> for our recommended system requirements.NGSeasy will run on any modern computer/workstation or cloud infrastructure.  The Hard Disk requirements are based on our experience and result from the fact that the pipeline/tools produce a range of intermediary and temporary files for each sample.</p>

<p>The full NGSeasy install includes indexed genomes for hg19 and b37 for all aligners, annotation files from GATK resource, and all of the NGSeasy docker images. Additional disk space is needed if the user wishes to install the databases associated with the variant annotators, Annovar, VEP and snpEff.</p>

<p>Based on our experience, a functional basic NGS compute system for a small lab, would consist of at least 4TB disk space, 60GB RAM and at least 32 CPU cores. Internet speed and network connectivity are a major bottle neck when dealing with NGS sized data, and groups are encouraged to think about these issues before embarking on multi sample or population level studies - where compute requirements can very quickly escalate.  </p>

<p><strong>System Requirements</strong></p>

<table>
<thead>
<tr>
<th>Component</th>
<th>Minimum</th>
<th>Recommended</th>
</tr>
</thead>
<tbody>
<tr>
<td>RAM</td>
<td>16GB</td>
<td>48-60GB</td>
</tr>
<tr>
<td>CPU</td>
<td>8 cores</td>
<td>16-36 cores</td>
</tr>
<tr>
<td>Hard Disk (per sample)</td>
<td>50-100GB</td>
<td>200-500GB</td>
</tr>
<tr>
<td>NGSeasy Install</td>
<td>200GB</td>
<td>500GB</td>
</tr>
<tr>
<td>Annotation Databases</td>
<td>500GB</td>
<td>&gt;1TB</td>
</tr>
</tbody>
</table>

<h2>
<a id="installing-docker" class="anchor" href="#installing-docker" aria-hidden="true"><span class="octicon octicon-link"></span></a>Installing Docker</h2>

<p>Follow the simple instructions in the links provided below  </p>

<ul>
<li>
<a href="https://docs.docker.com/installation/mac/">Mac</a><br>
</li>
<li><a href="https://docs.docker.com/installation/windows/">Windows</a></li>
<li><a href="https://docs.docker.com/installation/ubuntulinux/">Ubuntu</a></li>
</ul>

<p>A full set of instructions for multiple operating systems are available on the <a href="https://docs.docker.com/installation/">Docker website</a>.</p>

<h2>
<a id="getting-ngseasy" class="anchor" href="#getting-ngseasy" aria-hidden="true"><span class="octicon octicon-link"></span></a>Getting NGSeasy</h2>

<p>We provide a simple <strong>Makefile</strong> to pull all of the public nsgeasy components, scripts and set up to correct project directory structre on your local machines.</p>

<p>Setting up the initial project can take up a day, depending on your local network connections and speeds.</p>

<p>The default install dir is the users <strong>${HOME}</strong> directory.
The  <strong>Makefile</strong> provides options to install to any user defined directory and select NGSeasy version. eg :-</p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c">## EG. Installing to /media/scratch</span>
make INSTALLDIR=<span class="pl-s"><span class="pl-pds">"</span>/media/scratch<span class="pl-pds">"</span></span> VERSION=<span class="pl-s"><span class="pl-pds">"</span>1.0<span class="pl-pds">"</span></span> all</pre></div>

<p>The  <strong>Makefile</strong> also allows installation of selected components (check out its insides!).</p>

<h2>
<a id="set-up-ngseasy-project-configuration-file" class="anchor" href="#set-up-ngseasy-project-configuration-file" aria-hidden="true"><span class="octicon octicon-link"></span></a>Set up NGSeasy Project configuration file</h2>

<p>Using Excel or something, make a <strong>[config.file.tsv]</strong>  file and save as [TAB] a Delimited file with <code>.tsv</code> extenstion.
This sets up Information related to: Project Name, Sample Name, Library Type, Pipeline to call, NCPU.</p>

<p>We provide a template that can be used with NGSeasy, see <a href="https://docs.google.com/spreadsheets/d/13DosazuGdeAojZQ6YySM420p76RGleCYjR7a_MvcP2U/edit?usp=sharing">ngseasy_test.config.tsv</a>.</p>

<p>The <strong>[config.file.tsv]</strong> should contain the following 23 columns for each sample to be run through a pipeline:-</p>

<table>
<thead>
<tr>
<th>Variable</th>
<th>type</th>
<th>Description</th>
<th>Options(Examples)</th>
</tr>
</thead>
<tbody>
<tr>
<td>PROJECT_ID</td>
<td>STRING</td>
<td>Project ID</td>
<td>Cancer</td>
</tr>
<tr>
<td>SAMPLE_ID</td>
<td>STRING</td>
<td>Sample ID</td>
<td>SAMPLE_I</td>
</tr>
<tr>
<td>FASTQ1</td>
<td>STRING</td>
<td>Read 1 Fastq</td>
<td>foo_R1.fq.gz</td>
</tr>
<tr>
<td>FASTQ2</td>
<td>STRING</td>
<td>Read 2 Fastq</td>
<td>foo_R2.fq.gz</td>
</tr>
<tr>
<td>PROJECT_DIR</td>
<td>STRING</td>
<td>ngseasy project dir</td>
<td>/media/scratch/ngs_projects</td>
</tr>
<tr>
<td>DNA_PREP_LIBRARY_ID</td>
<td>STRING</td>
<td>NGS Library</td>
<td></td>
</tr>
<tr>
<td>NGS_PLATFORM</td>
<td>STRING</td>
<td>NGS Platform</td>
<td>ILLUMINA</td>
</tr>
<tr>
<td>NGS_TYPE</td>
<td>STRING</td>
<td>NGS Type</td>
<td>WEX (exome), WGS (genome), TGS (targeted)</td>
</tr>
<tr>
<td>BAIT</td>
<td>STRING</td>
<td>bait bed file</td>
<td>FOO.bed</td>
</tr>
<tr>
<td>CAPTURE</td>
<td>STRING</td>
<td>Capture bed file</td>
<td>BAR.bed</td>
</tr>
<tr>
<td>GENOMEBUILD</td>
<td>STRING</td>
<td>genome verison</td>
<td>hg19, b37 , b38 (coming soon)</td>
</tr>
<tr>
<td>FASTQC</td>
<td>STRING</td>
<td>Select fastqc</td>
<td>no-fastqc, qc-fastqc</td>
</tr>
<tr>
<td>TRIM</td>
<td>STRING</td>
<td>Select trimming</td>
<td>no-trimm, atrimm, btrimm</td>
</tr>
<tr>
<td>BSQR</td>
<td>STRING</td>
<td>Select BSQR</td>
<td>no-bsqr, bam-bsqr, gatk-bsqr</td>
</tr>
<tr>
<td>REALN</td>
<td>STRING</td>
<td>Select Realignment</td>
<td>no-realn, bam-realn, gatk-realn</td>
</tr>
<tr>
<td>ALIGNER</td>
<td>STRING</td>
<td>Select Aligner</td>
<td>no-aln, bwa, stampy, snap, novoalign, bowtie2</td>
</tr>
<tr>
<td>VARCALLER</td>
<td>STRING</td>
<td>Select Variant Caller</td>
<td>no-varcall, freebayes, platypus, UnifiedGenotyper, HaplotypeCaller, ensemble</td>
</tr>
<tr>
<td>CNV</td>
<td>STRING</td>
<td>Select CNV caller</td>
<td>no-sv,all-sv,lumpy,delly,slope,exomedepth,mhmm,cnvnator</td>
</tr>
<tr>
<td>ANNOTATOR</td>
<td>STRING</td>
<td>Select variant annotator</td>
<td>no-anno,snpeff,annovar,vep</td>
</tr>
<tr>
<td>CLEANUP</td>
<td>STRING</td>
<td>clean up temp files</td>
<td>TRUE, FALSE</td>
</tr>
<tr>
<td>NCPU</td>
<td>NUMBER</td>
<td>number of cores</td>
<td>1 .. N</td>
</tr>
<tr>
<td>VERSION</td>
<td>NUMBER</td>
<td>NGSeasy version</td>
<td>1.0</td>
</tr>
<tr>
<td>NGSUSER</td>
<td>STRING</td>
<td>user email</td>
<td><a href="mailto:stephen.j.newhouse@gmail.com">stephen.j.newhouse@gmail.com</a></td>
</tr>
</tbody>
</table>

<h2>
<a id="the-ngseasy-project-directory" class="anchor" href="#the-ngseasy-project-directory" aria-hidden="true"><span class="octicon octicon-link"></span></a>The NGSeasy project directory</h2>

<p>The user needs to make the relevent directory structures on their local machine before starting an NGS run.</p>

<p>On our sysetm we typically set up a top-level driectory called <strong>ngs_projects</strong> within which we store output from all our individual NGS projects.</p>

<p>Here we are working from local top level directory called <strong>media/</strong>, but this can really be any folder on your local system ie your home directory <strong>~/${USER}</strong>.  </p>

<p>Within this directory <strong>media</strong> we make the following folders: -</p>

<div class="highlight highlight-source-shell"><pre>ngs_projects  
<span class="pl-k">|</span>  
<span class="pl-k">|</span>__raw_fastq  
<span class="pl-k">|</span>__config_files  
<span class="pl-k">|</span>__ngseasy_resources  
   <span class="pl-k">|</span>  
   <span class="pl-k">|</span>__reference_genomes_b37  
   <span class="pl-k">|</span>__reference_genomes_hg19</pre></div>

<p>Running the script <code>make XXXX</code> ensures that all relevant directories are set up, and also enforces a clean structure to the NGS project.  </p>

<p>Within this we make a <code>raw_fastq</code> folder, where we temporarily store all the raw fastq files for each project. This folder acts as an initial stagging area for the raw fastq files. During the project set up, we copy/move project/sample related fastq files to their own specific directories.
Fastq files must have suffix and be gzipped: <strong>_1.fq.gz</strong> or <strong>_2.fq.gz</strong><br>
furture version will allow any format  </p>

<p>Running <code>ngseasy</code> with the relevent configuration file, will set up the following directory structure for every project and sample within a project:-  </p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c1">.</span>
ngs_projects  
<span class="pl-k">|</span>  
<span class="pl-k">|</span>__raw_fastq  
<span class="pl-k">|</span>__config_files  
<span class="pl-k">|</span>__run_logs
<span class="pl-k">|</span>__ngseasy_resources
<span class="pl-k">|</span>
<span class="pl-k">|</span>__ project_id  
    <span class="pl-k">|</span>  
    <span class="pl-k">|</span>__run_logs  
    <span class="pl-k">|</span>__config_files  
    <span class="pl-k">|</span>
    <span class="pl-k">|</span>__sample_id_1  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>__fastq  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>__tmp  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>__alignments  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>__vcf  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>__reports  
    <span class="pl-k">|</span>   <span class="pl-k">|</span>__config_files  
    <span class="pl-k">|</span>
    <span class="pl-k">|</span>
    <span class="pl-k">|</span>__sample_id_n  
        <span class="pl-k">|</span>  
        <span class="pl-k">|</span>__fastq  
        <span class="pl-k">|</span>__tmp  
        <span class="pl-k">|</span>__alignments  
        <span class="pl-k">|</span>__vcf  
        <span class="pl-k">|</span>__reports  
        <span class="pl-k">|</span>__config_files  </pre></div>

<h2>
<a id="the-raw_fastq-directory" class="anchor" href="#the-raw_fastq-directory" aria-hidden="true"><span class="octicon octicon-link"></span></a>The raw_fastq Directory</h2>

<p>The <code>raw_fastq</code> Directory is a very special directory indeed.
This is where the user should copy and or move <strong>ALL NEW RAW FASTQ</strong> Files to.
This is to be used as an intial staging area for all fastq files.
NGSeasy expects all raw fastq data to be placed here for all new samples or runs.
NGSeasy inspects this folder and looks for the fastq file names specified in your confifg file.
If NGSeasy doen't find them, then it exits.
We do this to force the user to get organised.</p>

<hr>

<h2>
<a id="manually-build-required-ngseasy-container-images" class="anchor" href="#manually-build-required-ngseasy-container-images" aria-hidden="true"><span class="octicon octicon-link"></span></a>Manually Build required NGSeasy Container Images</h2>

<p><strong>Work In Progress...</strong></p>

<p>Currently we are not able to automatically build some of the tools in pre-built docker containers due to licensing restrictions.</p>

<p>Some of the software has restrictions on use particularly for commercial
purposes. Therefore if you wish to use this for commercial purposes, then you
leagally have to approach the owners of the various components yourself!  </p>

<p><strong>Software composing the pipeline requiring registration:-</strong>  </p>

<ul>
<li>novoalign <a href="http://www.novocraft.com/">http://www.novocraft.com/</a><br>
</li>
<li>GATK <a href="https://www.broadinstitute.org/gatk/">https://www.broadinstitute.org/gatk/</a><br>
</li>
<li>ANNOVAR <a href="http://www.openbioinformatics.org/annovar/">http://www.openbioinformatics.org/annovar/</a><br>
</li>
</ul>

<p><strong>These tools require manual download and registration with the proivder. For non-academics/commercial groups, you will need to pay for some of these tools.</strong></p>

<p>Once you have paid/registered and downloaded the tool, we provide scripts and guidance for building these tools on your system.  </p>

<p>Its as easy as:-  </p>

<div class="highlight highlight-source-shell"><pre>docker build -t compbio/ngseasy-<span class="pl-smi">${TOOL}</span> <span class="pl-c1">.</span></pre></div>

<h2>
<a id="building-novoalign" class="anchor" href="#building-novoalign" aria-hidden="true"><span class="octicon octicon-link"></span></a>Building NOVOALIGN</h2>

<p><strong>Download Novoalign from  <a href="http://www.novocraft.com/">http://www.novocraft.com/</a></strong> into the local build directory *<em>ngseasy/containerized/ngs_docker_debian/ngs_aligners/ngseasy_novoalign</em>.
Edit the <a href="https://github.com/KHP-Informatics/ngseasy/blob/master/containerized/ngs_docker_debian/ngs_aligners/ngseasy_novoalign/Dockerfile">Dockerfile</a>
to relfect the correct version of novoalign.  </p>

<p>To use all novoalign fucntionality, you will need to <strong>pay for a license</strong>.</p>

<p>Once you obtained your <strong>novoalign.lic</strong>, download this to the build directory *<em>ngseasy/containerized/ngs_docker_debian/ngs_aligners/ngseasy_novoalign</em>, which now should contain your updated <a href="https://github.com/KHP-Informatics/ngseasy/blob/master/containerized/ngs_docker_debian/ngs_aligners/ngseasy_novoalign/Dockerfile">Dockerfile</a>.</p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c"># move to ngseasy_stampy folder</span>
<span class="pl-c1">cd</span> ngseasy/containerized/ngs_docker_debian/ngs_aligners/ngseasy_novoalign
ls</pre></div>

<p><strong>the directory should contain the following:-</strong></p>

<pre><code>Dockerfile
novoalign.lic
README.md
novosortV1.03.01.Linux3.0.tar.gz
novocraftV3.02.08.Linux3.0.tar.gz
</code></pre>

<p><strong>build novoalign</strong></p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c"># build</span>
docker build -t compbio/ngseasy-novoalign:v1.0 <span class="pl-c1">.</span></pre></div>

<hr>

<h2>
<a id="building-gatk" class="anchor" href="#building-gatk" aria-hidden="true"><span class="octicon octicon-link"></span></a>Building GATK</h2>

<p>You need to register and accept the GATK license agreement at <a href="https://www.broadinstitute.org/gatk/">https://www.broadinstitute.org/gatk/</a>.  </p>

<p>Once done, download GATK and place in the GTAK build directory <strong>ngseasy/containerized/ngs_docker_debian/ngs_utils/ngseasy_gatk</strong>.  </p>

<p>Edit the <a href="https://github.com/KHP-Informatics/ngseasy/blob/master/containerized/ngs_docker_debian/ngs_utils/ngseasy_gatk/Dockerfile">Dockerfile</a> to relfect
the correct version of GATK.  </p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c"># move to ngseasy_gatk folder</span>
<span class="pl-c1">cd</span> ngseasy/containerized/ngs_docker_debian/ngs_utils/ngseasy_gatk
ls</pre></div>

<p><strong>the directory should contain the following:-</strong></p>

<pre><code>Dockerfile
README.md
GenomeAnalysisTK-3.3-0.tar.bz2
</code></pre>

<p><strong>build gatk</strong></p>

<div class="highlight highlight-source-shell"><pre><span class="pl-c"># build</span>
docker build -t compbio/ngseasy-gatk:v1.0 <span class="pl-c1">.</span></pre></div>

<hr>

<h2>
<a id="manually-build-ngseasy-variant-annotaion-container-images" class="anchor" href="#manually-build-ngseasy-variant-annotaion-container-images" aria-hidden="true"><span class="octicon octicon-link"></span></a>Manually Build NGSeasy Variant Annotaion Container Images</h2>

<p>The tools used for variant annotation use large databases and the docker images exceed 10GB. Therefore, the user should manually build these container images prior to running the NGS pipelines.
Docker build files (<a href="https://docs.docker.com/jsearch/?q=Dockerfile">Dockerfile</a>) are available for</p>

<ul>
<li>
<a href="https://github.com/KHP-Informatics/ngs/tree/master/containerized/ngs_docker_debian/ngseasy_annovar/Dockerfile">Annovar</a><br>
</li>
<li><a href="https://github.com/KHP-Informatics/ngs/tree/master/containerized/ngs_docker_debian/ngseasy_vep/Dockerfile">VEP</a></li>
<li>
<a href="https://github.com/KHP-Informatics/ngs/tree/master/containerized/ngs_docker_debian/ngseasy_snpeff/Dockerfile">snpEff</a><br>
</li>
</ul>

<p><strong>Note</strong> Annovar requires user registration.  </p>

<p><strong>Once built on the user system, these container images can persist for as long as the user wants.</strong>  </p>

<p><strong>Large Variant Annotation Container Images</strong></p>

<p>Its as easy as:-  </p>

<div class="highlight highlight-source-shell"><pre>docker build -t compbio/ngseasy-<span class="pl-smi">${TOOL}</span> <span class="pl-c1">.</span></pre></div>

<hr>

<h2>
<a id="build-vep" class="anchor" href="#build-vep" aria-hidden="true"><span class="octicon octicon-link"></span></a>Build VEP</h2>

<div class="highlight highlight-source-shell"><pre>
<span class="pl-c1">cd</span> /media/ngs_projects/nsgeasy/ngs/containerized/ngs_docker_debian/ngseasy_vep

sudo docker build -t compbio/ngseasy-vep:<span class="pl-smi">${VERSION}</span> <span class="pl-c1">.</span></pre></div>

<hr>

<h2>
<a id="build-annovar" class="anchor" href="#build-annovar" aria-hidden="true"><span class="octicon octicon-link"></span></a>Build Annovar</h2>

<div class="highlight highlight-source-shell"><pre><span class="pl-c1">cd</span> /media/ngs_projects/nsgeasy/ngs/containerized/ngs_docker_debian/ngseasy_annovar

sudo docker build -t compbio/ngseasy-annovar:<span class="pl-smi">${VERSION}</span> <span class="pl-c1">.</span></pre></div>

<hr>

<h2>
<a id="build-snpeff" class="anchor" href="#build-snpeff" aria-hidden="true"><span class="octicon octicon-link"></span></a>Build snpEff</h2>

<div class="highlight highlight-source-shell"><pre><span class="pl-c1">cd</span> /media/ngs_projects/nsgeasy/ngs/containerized/ngs_docker_debian/ngseasy_snpeff

sudo docker build -t compbio/ngseasy-snpeff:<span class="pl-smi">${VERSION}</span> <span class="pl-c1">.</span></pre></div>

<hr>

<h2>
<a id="coming-soon" class="anchor" href="#coming-soon" aria-hidden="true"><span class="octicon octicon-link"></span></a>Coming Soon</h2>

<ul>
<li>New Aligners:- <a href="http://snap.cs.berkeley.edu/">SNAP</a>, GSNAP, mr- and mrs-Fast,gem</li>
<li><a href="https://github.com/amplab/snap">https://github.com/amplab/snap</a></li>
<li>SLOPE (CNV fo targetted NSG)</li>
<li>Cancer Pipelines</li>
<li>Annotation Pipelines and Databases</li>
<li>Visualisation Pipelines</li>
<li>Var Callers:- VarScan2</li>
<li>SGE scripts and basic BASH scrips for running outside of Docker</li>
<li>biobambam <a href="https://github.com/gt1/biobambam">https://github.com/gt1/biobambam</a><br>
</li>
<li>bamaddrg <a href="https://github.com/ekg/bamaddrg">https://github.com/ekg/bamaddrg</a><br>
</li>
<li>bamtools <a href="https://github.com/ekg/bamtools">https://github.com/ekg/bamtools</a><br>
</li>
</ul>

<h2>
<a id="useful-links" class="anchor" href="#useful-links" aria-hidden="true"><span class="octicon octicon-link"></span></a>Useful Links</h2>

<ul>
<li>
<a href="https://bcbio.wordpress.com/">https://bcbio.wordpress.com/</a><br>
</li>
<li>
<a href="https://basecallbio.wordpress.com/2013/04/23/base-quality-score-rebinning/">https://basecallbio.wordpress.com/2013/04/23/base-quality-score-rebinning/</a><br>
</li>
<li>
<a href="https://github.com/statgen/bamUtil">https://github.com/statgen/bamUtil</a><br>
</li>
<li>
<a href="http://genome.sph.umich.edu/wiki/BamUtil:_recab">http://genome.sph.umich.edu/wiki/BamUtil:_recab</a><br>
</li>
<li>
<a href="https://github.com/chapmanb/bcbio.variation">https://github.com/chapmanb/bcbio.variation</a><br>
</li>
<li><a href="http://plagnol-lab.blogspot.co.uk/2013/11/faq-and-clarifications-for-exomedepth-r.html">http://plagnol-lab.blogspot.co.uk/2013/11/faq-and-clarifications-for-exomedepth-r.html</a></li>
</ul>

<hr>

<p>(C) 2015 Stephen J Newhouse &amp; Amos Folarin</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/KHP-Informatics/ngseasy">NGSeasy</a> is maintained by <a href="https://github.com/KHP-Informatics">KHP-Informatics</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
